xikolo-ios/AppDelegate.swift

```diff
@@ -14,9 +14,27 @@ class AppDelegate : AbstractAppDelegate {
 
     override func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool {
         window?.tintColor = Brand.TintColor
+        updateNews()
+
         return super.application(application, didFinishLaunchingWithOptions: launchOptions)
     }
 
+    func updateNews() {
+        NewsArticleHelper.syncNewsArticles().onSuccess { (newsArticles) in // sync news and show badge on news tab with number of unread articles
+            if let rootViewController = self.window?.rootViewController as? UITabBarController {
+                if let tabArray = rootViewController.tabBar.items {
+                    let tabItem = tabArray[2]
```

Comment:
Is it possible to improve the detection of the "news" tab? This seems error-prone to me.